compute_tools/src/compute.rs

```diff
@@ -1937,19 +1937,46 @@ LIMIT 100",
         tokio::spawn(conn);
 
         // TODO: support other types of grants apart from schemas?
-        let query = format!(
-            "GRANT {} ON SCHEMA {} TO {}",
-            privileges
-                .iter()
-                // should not be quoted as it's part of the command.
-                // is already sanitized so it's ok
-                .map(|p| p.as_str())
-                .collect::<Vec<&'static str>>()
-                .join(", "),
-            // quote the schema and role name as identifiers to sanitize them.
-            schema_name.pg_quote(),
-            role_name.pg_quote(),
-        );
+
+        // check the role grants first - to gracefully handle read-replicas.
+        let select = "SELECT privilege_type
+            FROM pg_namespace
+                JOIN LATERAL (SELECT * FROM aclexplode(nspacl) AS x) acl ON true
+                JOIN pg_user users ON acl.grantee = users.usesysid
+            WHERE users.usename = $1
+                AND nspname = $2";
+        let rows = db_client
+            .query(select, &[role_name, schema_name])
```

Comment:
Do we need to `role_name.pg_quote()` here as well? I didn't see notes of quotations in `.query()`: https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Client.html#method.query

Reply:
No. `pg_quote` is for identifier quoting, eg it uses `"` in the query. That's correct for the grant query but not for this lookup which needs query literals. The query method internally uses prepared statements so doesn't need any quoting regardless.